File: Lab/stack.c
// stack representation
#include<stdio.h>
#include<stdlib.h>
#define size 25
void push(int ele,int *top,int stack[])
{
    if(*top == size)
    {
        printf("stack full!\n");
        return;
    }
    else
    {
        stack[++(*top)] = ele;
        return;
    }
}

void pop(int *top,int stack[])
{
    if(*top == -1)
    {
        printf("stack empty\n");
        return;
    }
    else
        printf("%d has been popped\n",stack[(*top)--]);
}

void display(int top,int stack[])
{
    if(top == -1)
    {
        printf("stack empty\n");
        return;
    }
    else
    {
        for(int i=top;i>=0;i--)
        {
            printf("%d\n",stack[i]);
        }
    }
}

void main()
{
    int top = -1;
    int stack[size],ch,ele;
    for(;;)
    {
        printf("Enter\n1. to push\n2. to pop\n3. to display\n4. exit\n");
        scanf("%d",&ch);
        switch(ch)
        {
            case 1 :
            printf("enter element to be pushed : ");
            scanf("%d",&ele);
            push(ele,&top,stack);
            break;
            case 2 :
            pop(&top,stack);
            break;
            case 3 :
            display(top,stack);
            break;
            case 4 :
            exit(0);
            default : printf("Enter input from (1-4) only!!\n");
            break;
        }
    }
}

File: 4th SEM Lab/java/stack.java
import java.util.Scanner;
public class stack {
    public static Scanner scan = new Scanner(System.in);
    static int top = -1;
    static int arr[] = new int[10];

    static void print(Object line){
        System.out.print(line);
    }
    static void println(Object line){
        System.out.println(line);
    }
    void push(){
        if(top >= arr.length){
            println("Stack full");
            return;
        }
        print("Enter element to be pushed : ");
        int ele = scan.nextInt();
        arr[++top] = ele;
    }

    void pop(){
        if(top == -1){
            println("Stack empty");
            return;
        }
        println(""+arr[top--]);
    }

    void display(){
        for(int i = top;i>=0;i--){
            println(""+arr[i]);
        }
    }
    public static void main(String args[]){
        stack st = new stack();
        while(true){
            println("Enter\n1.Push\n2.Pop\n3.Display\n4.exit");
            int choice = scan.nextInt();
                switch(choice){
                    case 1 : st.push();
                            break;
                    case 2 : st.pop();
                    break;
                    case 3 :st.display();
                    break;
                    case 4 : System.exit(0);  
                }

        }
        


    }
    
}
/*
Enter
1.Push
2.Pop
3.Display
4.exit
1
Enter element to be pushed : 14
Enter
1.Push
2.Pop
3.Display
4.exit
1
Enter element to be pushed : 15
Enter
1.Push
2.Pop
3.Display
4.exit
2
15
Enter
1.Push
2.Pop
3.Display
4.exit
1
Enter element to be pushed : 13
Enter
1.Push
2.Pop
3.Display
4.exit
3
13
14
Enter
1.Push
2.Pop
3.Display
4.exit
4

 */